edgelet/edgelet-http/src/util/mod.rs

```diff
@@ -52,32 +57,59 @@ impl StreamSelector {
 
 impl Read for StreamSelector {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        match *self {
-            StreamSelector::Tcp(ref mut stream) => stream.read(buf),
+        match self {
+            StreamSelector::Tcp(stream) => return stream.read(buf),
+            StreamSelector::TlsConnecting(stream) => match stream.poll() {
+                Ok(Async::Ready(stream)) => {
+                    *self = StreamSelector::TlsConnected(stream);
+                    return self.read(buf);
+                }
+                Ok(Async::NotReady) => return Err(ErrorKind::WouldBlock.into()),
+                Err(e) => return Err(io::Error::new(ErrorKind::Other, e)),
+            },
+            StreamSelector::TlsConnected(stream) => return stream.read(buf),
             #[cfg(windows)]
-            StreamSelector::Pipe(ref mut stream) => stream.read(buf),
-            StreamSelector::Unix(ref mut stream) => stream.read(buf),
-        }
+            StreamSelector::Pipe(stream) => return stream.read(buf),
+            StreamSelector::Unix(stream) => return stream.read(buf),
+        };
```

Comment:
Instead of `;` and explicit `return`s, just remove both.

Same for the other functions.